src/jit/emitarm64.cpp

```diff
@@ -7977,120 +8101,290 @@ BYTE*               emitter::emitOutputLJ(insGroup  *ig, BYTE *dst, instrDesc *i
+    }
+
+    dst += emitOutput_Instr(dst, code);
+
+    return dst;
+}
+
+/*****************************************************************************
+*
+*  Output a short constant instruction.
+*/
+BYTE*               emitter::emitOutputShortConstant(BYTE *dst, instruction ins, insFormat fmt, ssize_t imm, regNumber reg, emitAttr opSize)
+{
+    code_t code = emitInsCode(ins, fmt);
+
+    if (fmt == IF_LS_1A)
+    {
+        // LS_1A   XX...V..iiiiiiii iiiiiiiiiiittttt      Rt simm21
+        // INS_ldr or INS_ldrsw (PC-Relative)
+
+        ssize_t loBits = (imm & 3);
+        noway_assert(loBits == 0);
+        ssize_t distVal = imm >>= 2; // load offset encodings are scaled by 4.
+
+        noway_assert(isValidSimm19(distVal));
+
+        // Is the target a vector register?
+        if (isVectorRegister(reg))
+        {
+            code |= insEncodeDatasizeVLS(code, opSize);  // XX V
+            code |= insEncodeReg_Vt(reg);                // ttttt
+        }
+        else
+        {
+            assert(isGeneralRegister(reg));
+            // insEncodeDatasizeLS is not quite right for this case.
+            // So just specialize it.
+            if ((ins == INS_ldr) && (opSize == EA_8BYTE))
+            {
+                // set the operation size in bit 30
+                code |= 0x40000000;
             }
 
-            noway_assert(loBits == 0);
-            noway_assert(isValidSimm19(distVal));
-            distVal &= 0x7FFFFLL;                         // 19 bits 
-            code |= distVal <<  5;
+            code |= insEncodeReg_Rt(reg);                // ttttt
         }
-        else if (fmt == IF_DI_1E)  // DI_1E   .ii.....iiiiiiii iiiiiiiiiiiddddd      Rd simm21
+
+        distVal &= 0x7FFFFLL;                            // 19 bits
+        code |= distVal << 5;
+    }
+    else if (fmt == IF_LS_2B)
+    {
+        //  ldr     Rt,[Xn+pimm12]       LS_2B  1X11100101iiiiii iiiiiinnnnnttttt   B940 0000   imm(0-4095<<{2,3})
+        // INS_ldr or INS_ldrsw (PC-Relative)
+        noway_assert(isValidUimm12(imm));
+        assert(isGeneralRegister(reg));
+
+        // insEncodeDatasizeLS is not quite right for this case.
+        // So just specialize it.
+        if ((ins == INS_ldr) && (opSize == EA_8BYTE))
         {
-            // INS_adr or INS_adrp
-            code |= insEncodeReg_Rd(id->idReg1());        // ddddd
+            // set the operation size in bit 30
+            code |= 0x40000000;
+        }
 
-            noway_assert(isValidSimm19(distVal));
-            distVal &= 0x7FFFFLL;                         // 19 bits 
-            code |= distVal <<  5;
-            code |= loBits  << 29;                        //  2 bits
+        if (opSize == EA_8BYTE)
+        {
+            // insEncodeDatasizeLS is not quite right for this case.
+            // So just specialize it.
+            if (ins == INS_ldr)
+            {
+                // set the operation size in bit 30
+                code |= 0x40000000;
```

Comment:
This seems redundant as we just set the bit using the code above.
One of these two blocks can be removed